refactor(commitlint): change `scope-case` rules

pawelkmpt opened the pull request:
Introduced in #4 

Discussion https://cxlworld.slack.com/archives/C01JABH8AHX/p1645522633448099

Using `never` for `kebab-case`, `snake-case` causes errors for lowercase, one word scopes, eg. `composer`, `phpcs` which we use a lot. `always` rules seem to be better idea.

pawelkmpt committed:
refactor(commitlint): change `scope-case` rules

Using 'never' for 'kebab-case', 'snake-case' causes errors for
lowercase, one word scopes, eg. composer, phpcs.

pawelkmpt commented:
I keep testing on other repos and I think we should skip validation of scope, because of:
* `refactor(widgets/LastUpdated)`,
* `refactor(JS/fastcomments)`.

Also subject case validation doesn't make sense IMO. We like to put `method_name()`, `ClassName` or `package-name` and each of these examples causes lint throw errors.

I think most important thing is to have correct commit type (`feat,` `chore`, etc.) and not empty subject.

Your thoughts @lkraav @saas786

saas786 commented:
> I keep testing on other repos and I think we should skip validation of scope, because of:
> 
> * `refactor(widgets/LastUpdated)`,
> * `refactor(JS/fastcomments)`.
> 
> Also subject case validation doesn't make sense IMO. We like to put `method_name()`, `ClassName` or `package-name` and each of these examples causes lint throw errors.
> 
> I think most important thing is to have correct commit type (`feat,` `chore`, etc.) and not empty subject.
> 
> Your thoughts @lkraav @saas786

I am up for anything, that allows me to be consistent, follow standards for commit msgs.

Yes, type & message are most important.

Scope can be skipped, as it can be combination of things. So limiting scope makes it hard, like in case of headroom.

had to use `cxl-lumo-styles`, where either `headroom` or `sticky-header` or `top-header` could make much more sense to me.

lkraav commented:
> had to use cxl-lumo-styles, where either headroom or sticky-header or top-header could make much more sense to me.

`type(scope): adjust headroom implementation etc` can always be written, such that you don't need to vary scope so much. When browsing commit log, understanding what top-level scope was modified should make understanding following commit message context easier.

> I keep testing on other repos and I think we should skip validation of scope, because of:

We should enable it globally, on principle, and potentially override in repos that currently have trouble with it.

BUT seeing errors with past commit messages doesn't really matter much, as we can change convention to write commit messages with any new strategy at any time.

Question is, how to best define the list of scopes for each repo, and that really can't be done centrally (I think), so needs to be a repo-specific issue.

saas786 commented:
Certainly, it can't be defined globally, enabled yes, but scope is local.

Does `scope` covers specific package? or `classname` or `feature` etc? How we define a fine boundary or scope for `scope` :)

pawelkmpt committed:
refactor(commitlint): change `subject-case` rules

Comparing to upstream 'sentence-case' was removed to allow first word
starting with capital letter, eg. "WooCommerce foo bar".

Real commit message example which is passing now but would not pass before:
refactor(Integrations): Sensei display playbooks related content

lkraav commented:
I added Slack discussion link to opening comment.
